src/corefx/System.Globalization.Native/localeNumberData.cpp

```diff
@@ -31,9 +47,259 @@ enum LocaleNumberData : int32_t
 
 /*
 Function:
+NormalizePattern
+
+Returns a numeric string pattern in a format that we can match against the appropriate managed pattern.
+*/
+UnicodeString* NormalizePattern(const UnicodeString *srcPattern, UnicodeString *destPattern, bool isNegative)
+{
+	// A srcPattern example: "#,##0.00 C;(#,##0.00 C)" but where C is the international currency symbol (chCurrencySign)
+	// The positive pattern comes first, then an optional negative pattern separated by a semicolon
+	// A destPattern example: "(C n)" where C represents the currency symbol, and n is the number
+	destPattern->remove();
+
+	int iStart = 0;
+	int iEnd = srcPattern->length() - 1;
+	int32_t iNegative = srcPattern->indexOf(chPatternSeparator);
+	if (iNegative >= 0)
+	{
+		if (isNegative)
+		{
+			iStart = iNegative + 1;
+		}
+		else
+		{
+			iEnd = iNegative - 1;
+		}
+	}
+
+	bool minusAdded = false;
+	bool digitAdded = false;
+	bool currencyAdded = false;
+	bool spaceAdded = false;
+
+	for (int i = iStart; i <= iEnd; i++)
+	{
+		UChar32 ch = srcPattern->char32At(i);
+		switch (ch)
+		{
+		case chPatternDigit:
+			if (!digitAdded)
+			{
+				digitAdded = true;
+				destPattern->append('n');
+			}
+			break;
+
+		case chCurrencySign:
+			if (!currencyAdded)
+			{
+				currencyAdded = true;
+				destPattern->append('C');
+			}
+			break;
+
+		case chSpace:
+		case chSpace2:
+			if (!spaceAdded)
+			{
+				spaceAdded = true;
+				destPattern->append(chSpace);
+			}
+			else
+			{
+				assert(false);
+			}
+			break;
+
+		case chPatternMinus:
+		case chOpenParen:
+		case chCloseParen:
+			minusAdded = true;
+			destPattern->append(ch);
+			break;
+
+		case chPercent:
+			destPattern->append(ch);
+			break;
+		}
+	}
+
+	// if there is no segative subpattern, the convention is to prefix the minus sign
+	if (isNegative && !minusAdded)
+	{
+		destPattern->insert(0, chPatternMinus);
+	}
+
+	return destPattern;
+}
+
+/*
+Function:
+GetPattern
+
+Determines the pattern from the decimalFormat and returns the matching pattern's index from patterns[].
+Returns index -1 if no pattern is found.
+*/
+int GetPattern(DecimalFormat *decimalFormat, const char* patterns[], int patternsCount, bool isNegative)
+{
+	const int MAX_DOTNET_NUMERIC_PATTERN_LENGTH = 6; // example: "(C n)" plus terminator
+	char charPattern[MAX_DOTNET_NUMERIC_PATTERN_LENGTH] = { 0 };
+
+	UnicodeString icuPattern;
+	icuPattern.remove();
```

Comment:
No it does not. Good catch. The next line to get the pattern clears the string anyway